Add Datastore Automation Rules (#31835)
DataDog/documentation: content/en/actions/datastores/trigger.md

---
title: Use Automation Rules 
description: Reduce manual effort by triggering workflows automatically when datastore records are created, updated, or change status. 
disable_toc: false
aliases:
- actions/datastore/trigger
further_reading:
- link: "https://www.datadoghq.com/blog/datadog-datastore/"
  tag: "Blog"
  text: "Enhance your automated workflows and apps with Datastore"
---

> You can add workflows to your datastores that automatically trigger when records are created, updated, deleted

Author: ```suggestion
You can add workflows to your datastores that automatically trigger when records are created, updated, or deleted.
```

## Create a new automation rule

To create a new automation rule: 
1. Navigate to the [Datastores page][1].
1. Select a datastore. 
1. Click **Automation Rules**, then **+ New Automation Rule**. 
1. Configure your new automation rule: 
    1. Choose if the rule will be triggered when a new entry is added, a value has changed, or an entry is deleted.
    1. Choose the workflow that will be triggered when the event occurs. 
    1. Name your rule. 
1. Click **Save**.

## Edit an automation rule 

To edit an automation rule: 
1. Navigate to the [Datastores page][1].
1. Select a datastore. 
1. Click **Automation Rules**, then **+ New Automation Rule**. 
1. Click the **Edit** {{< img src="icons/pencil.png" inline="true" style="width:14px;">}} icon.
1. Make your changes. 
1. Click **Save Changes**.

## Delete an automation rule

To delete an automation rule: 
1. Navigate to the [Datastores page][1].
1. Select a datastore. 
1. Click **Automation Rules**, then **+ New Automation Rule**. 
1. Click the **Delete** {{< img src="icons/delete.png" inline="true" style="width:14px;">}} icon. 

> 1. Confirm deletion when prompted

Author: ```suggestion
1. When prompted, confirm deletion.
```

---
title: Use Automation Rules 
description: Reduce manual effort by triggering workflows automatically when datastore records are created, updated, or change status. 
disable_toc: false
aliases:
- actions/datastore/trigger
further_reading:
- link: "https://www.datadoghq.com/blog/datadog-datastore/"
  tag: "Blog"
  text: "Enhance your automated workflows and apps with Datastore"
---

> You can add workflows to your datastores that automatically trigger when records are created, updated, or change status.

Reviewer: I'm not sure what "change status" means, but I think it might be more accurate to say:

```suggestion
You can add workflows to your datastores that automatically trigger when records are created, updated, deleted
```

## Create a new automation rule

To create a new automation rule: 
1. Navigate to the [Datastores page][1].
1. Select a datastore. 
1. Click **Automation Rules**, then **+ New Automation Rule**. 
1. Configure your new automation rule: 
    1. Choose if the rule will be triggered when a new entry is added, a value has changed, or an entry is deleted.
    1. Choose the workflow that will be triggered when the event occurs. 
    1. Name your rule. 
1. Click **Save**.

## Edit an automation rule 

To edit an automation rule: 
1. Navigate to the [Datastores page][1].
1. Select a datastore. 
1. Click **Automation Rules**, then **+ New Automation Rule**. 
1. Click the **Edit** {{< img src="icons/pencil.png" inline="true" style="width:14px;">}} icon.
1. Make your changes. 
1. Click **Save Changes**.

## Delete an automation rule

To delete an automation rule: 
1. Navigate to the [Datastores page][1].
1. Select a datastore. 
1. Click **Automation Rules**, then **+ New Automation Rule**. 
1. Click the **Delete** {{< img src="icons/delete.png" inline="true" style="width:14px;">}} icon. 

> 1. _Are you sure?_ will appear in the Datadog UI.
> 1. Click **Delete**.

Reviewer: If this is abiding by a convention, ignore me.

I think it might be more succinct to say:

```suggestion
1. Confirm deletion when prompted
```
